src/room_member.cpp

```diff
@@ -233,9 +234,13 @@ void RoomMember::ReceiveLoop() {
             peer->DeallocatePacket(packet);
         }
     }
+    peer->Shutdown(300);
```

Comment:
That's a problem how IsConnected is working, and not an issue of this PR. IsConnected only checks the state of the member. Since the member can still be connected but the state is already changed, this will result in a false response  of isConnected. 

Maybe it's better to check the state in this loop directly and create an issue for IsConnected. But this undefined state for isConnected will only exist for a few µs since as soon as isConnected will return false peer->Shutdown() is called.